### PR TITLE
Add OVAL limitations and pitfalls documentation section

### PR DESCRIPTION
**Pull request by yuumasato**
#### Description:

- Updated a bit the Checks OVAL section
- Added a new section to document OVAL limitations and traps, to document types of checks or situations where OVAL may fall short.

#### Rationale:

- Writing OVAL checks is not simple and sometimes it doesn't work the way one expects.

### CODE REVIEW

**Comment by yuumasato** [comment]
@openscap-ci test this please

File: docs/manual/developer/06_contributing_with_content.md [hunk 1]
```diff
@@ -610,15 +610,22 @@ Tips:
 
 ### Checks
 
-Checks are used to evaluate a Rule. They are written using a custom OVAL
-syntax and are stored as xml files inside the *checks/oval* directory
-for the desired platform. During the building process, the system will
-transform the checks in OVAL compliant checks.
+Checks are used to evaluate a Rule.
+They are written using a custom OVAL syntax and during the building process
+the system will transform them into OVAL compliant checks.
```
Comment (ggbecker) [inline docs/manual/developer/06_contributing_with_content.md:615]:
```suggestion
They are written using a custom OVAL syntax and transformed by the system
during the building process into OVAL compliant checks.
```

File: docs/manual/developer/06_contributing_with_content.md [hunk 2]
```diff
@@ -610,15 +610,22 @@ Tips:
...
+The OVAL checks are stored as XML files and the build system can source
+them from:
+
+- the `oval` directory of a rule
+- the shared pool of checks `shared/checks/oval`
+- a template
+
+In order to create a new check you must create a file in the
+appropriate directory. The *id* attribute of the check needs to match the *id*
+of the its rule.
```
Comment (ggbecker) [inline docs/manual/developer/06_contributing_with_content.md:626]:
```suggestion
of its rule.
```

File: docs/manual/developer/06_contributing_with_content.md [hunk 3]
```diff
@@ -701,6 +708,42 @@ oval_config_file_exists_test
 oval_config_file_exists_object
 ```
 
+#### Limitations and pitfalls
+
+This section aims to list known OVAL limitations and sutations that OVAL can't
```
Comment (ggbecker) [inline docs/manual/developer/06_contributing_with_content.md:713]:
```suggestion
This section aims to list known OVAL limitations and situations that OVAL can't
```

**Review by JAORMX — APPROVED** [review]
Wow! This is good to know.

**Review by ggbecker — CHANGES_REQUESTED** [review]
Few nitpicks in the text

File: docs/manual/developer/06_contributing_with_content.md (hunk 1 above)
Comment (yuumasato) [reply docs/manual/developer/06_contributing_with_content.md]: Thanks, I've edited your suggestion a bit applied it.

**Comment by yuumasato** [comment]
> Wow! This is good to know.

I was quite surprised when I found out about this.

This came out of a discussion with @cipherboy and deafpool on IRC.

**Comment by JAORMX** [comment]
/retest

The test had succeeded but subsequently CI failed posting the logs:

```
 INFO[2021-04-09T09:17:18Z] Running step e2e-aws-ocp4-moderate-test.     
INFO[2021-04-09T09:34:58Z] Step e2e-aws-ocp4-moderate-test succeeded after 17m40.057399326s. 
INFO[2021-04-09T09:34:58Z] Running step e2e-aws-ocp4-moderate-gather-aws-console. 
INFO[2021-04-09T09:35:18Z] Step e2e-aws-ocp4-moderate-gather-aws-console succeeded after 20.049578548s. 
INFO[2021-04-09T09:35:18Z] Running step e2e-aws-ocp4-moderate-gather-must-gather. 
INFO[2021-04-09T09:38:08Z] Step e2e-aws-ocp4-moderate-gather-must-gather succeeded after 2m50.05287059s. 
INFO[2021-04-09T09:38:08Z] Running step e2e-aws-ocp4-moderate-gather-extra. 
INFO[2021-04-09T09:38:18Z] Logs for container test in pod e2e-aws-ocp4-moderate-gather-extra: 
INFO[2021-04-09T09:38:18Z] Gathering artifacts ...                      
INFO[2021-04-09T09:38:18Z] The connection to the server api.**************-1708d.origin-ci-int-aws.dev.rhcloud.com:6443 was refused - did you specify the right host or port? 
INFO[2021-04-09T09:38:18Z] {"component":"entrypoint","error":"wrapped process failed: exit status 1","file":"prow/entrypoint/run.go:80","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","severity":"error","time":"2021-04-09T09:38:13Z"} 
INFO[2021-04-09T09:38:18Z] error: failed to execute wrapped command: exit status 1 
INFO[2021-04-09T09:38:18Z]                                              
```

**Comment by jan-cerny** [comment]
I think we can merge it for new, it's very useful.

**Comment by yuumasato** [comment]
@jan-cerny Thank you for review.


> I think we can merge it for new, it's very useful.

What do you mean by "new"?

**Pull request merged**